File: demo/src/main/java/com/utcn/demo/entity/Question.java
package com.utcn.demo.entity;


import jakarta.persistence.*;

@Entity
@Table(name = "question")
public class Question {

    @Id
    @Column(name="id")
    private Long questionId;

    @Column(name="title")
    private String title;


    @OneToOne
    @JoinColumn(name="c_id",nullable = false)
    private Content questionContent;

    public Question(){

    }

    public Question(Long questionId, String title, Content content) {
        this.questionId = questionId;
        this.title = title;
        this.questionContent = content;
    }

    public Long getQuestionId() {
        return questionId;
    }

    public void setQuestionId(Long questionId) {
        this.questionId = questionId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Content getQuestionContent() {
        return questionContent;
    }

    public void setQuestionContent(Content content) {
        this.questionContent = content;
    }
}

File: app/pages/question/question.ts
import {Content} from "../content";

export class Question {

  private _questionId:number;

  private _title:string;

  private _questionContent:Content;


  constructor(questionId: number, title: string, questionContent: Content) {
    this._questionId = questionId;
    this._title = title;
    this._questionContent = questionContent;
  }



  get questionId(): number {
    return this._questionId;
  }

  set questionId(value: number) {
    this._questionId = value;
  }

  get title(): string {
    return this._title;
  }

  set title(value: string) {
    this._title = value;
  }

  get questionContent(): Content {
    return this._questionContent;
  }

  set questionContent(value: Content) {
    this._questionContent = value;
  }
}
